state/protocol/badger/mutator.go

```diff
@@ -378,6 +382,12 @@ func (m *FollowerState) headerExtend(ctx context.Context, candidate *flow.Block,
 		if err != nil {
 			return fmt.Errorf("could not store candidate block: %w", err)
 		}
+		if proposerSigData != nil {
+			err = m.sigs.StoreTx(blockID, proposerSigData)(tx)
```

Comment:
We can still use a dedicated storage for proposal sigs, but I would suggest to use it pretty much the same way we use storage for payloads. To be explicit, we inject it in the `badger.Blocks` and use it as part of the transaction when storing the block.